keymanapp/keyman
# fix(android): Mirror arrow icons to support RTL

This PR creates 4 drawable layouts so that the forward/back navigation arrows and text size up/down arrows will autoMirror for RTL layouts.
* ic_action_back
* ic_action_forward
* ic_action_textsize_up
* ic_action_textsize_down

Base branch set to `chore/common/crowdin-ar` to facilitate testing with the Arabic locale (#5706)

### Sample screenshots when Display Language in Arabic (RTL layout)
language settings (forward arrow pointing left)
![Screenshot_1631928418](https://user-images.githubusercontent.com/7358010/133867843-24458031-89b6-4790-b69c-583e404a4ff3.png)

Text Size menu (increase text size pointing left, decrease text size pointing right)
![Screenshot_1631928435](https://user-images.githubusercontent.com/7358010/133867859-d603ebef-fd9f-4245-ac5d-717f464a7808.png)

Info activity (order of navigation icons: close, forward, back)
![Screenshot_1631928469](https://user-images.githubusercontent.com/7358010/133867874-ca16f6eb-0679-4728-9526-19d11dfe37d4.png)

## User Testing
Run the PR build on an Android 5.0 emulator

* **TEST_LTR ** - Tests no regressions in English UI layout
1. Load Keyman for Android and use the default English UI(left to right)
2. From the main menu, go to the Text Size menu
3. Use the menu and verify:
    * clicking left arrow decreases the text size
    * clicking right arrow increases the text size
4. From the  main menu, select Info Activity 
5. Browse through the help and verify
    * clicking left arrow will navigate back
    * clicking right arrow will navigate forward (only enabled if there's a forward page to go to)
6. Click "x" to close the Info activity
7. From the Settings menu, install sil_cameroon_qwerty keyboard from keyman.com for any language
8. Verify keyboard installation wizard behaves as expected
9. Click through other "Settings" menus and verify
    * settings toolbar back arrow points to the left (behaves as normal)
    * forward arrows point to the right (behaves as normal)

* **TEST_RTL ** - Test layout of RTL icons
1. Load Keyman for Android
2. From the "Settings" menu, change the Display language to "Arabic" (an RTL layout). Note that forward arrows will point left and back arrows will point right.
3. From the main menu, go to the Text Size menu
4. Use the menu and verify:
    * clicking the left arrow increases the text size 
    * clicking the right arrow decreases the text size
5. Browse through the help and verify
    * clicking left arrow will navigate forward (only enabled if there's a forward page to go to)
    * clicking right arrow will navigate back
6. Click "x" to close the Info activity
7. From the Settings menu, install sil_cameroon_qwerty keyboard from keyman.com for any language
8. Verify keyboard installation wizard behaves as expected. Note the bottom left buttons go "forward" in the wizard, and the "OK" buttons will be on the bottom left.
9. Click through other "Settings" menu and verify
    * settings toolbar back arrow points to the right
    * forward arrows point to the left
